### DISCUSSION TITLE
Caps lock not behaving

### DISCUSSION THREAD

**Discussion by Honestpuck** [pqrs-org/KE-complex_modifications#1154] [Q&A] [answered]
I'm trying to get Caps lock to control-option-command and nothing if pressed alone and both shifts together becoming caps lock.

I have it all working except caps lock is caps lock when pressed alone. I have attached the JOSN of my rules section. I've read the documentation backwards and forwards and I am missing something, any help would be appreciated. 

[karabiner.txt](https://github.com/pqrs-org/KE-complex_modifications/files/9017408/karabiner.txt)
.

**Comment by MuhammedZakir** [comment, marked as answer]
1. For "simultaneously pressing both keys -> caps_lcok" rule, `from.simultaneous` is better.
    - https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/from/simultaneous/
    - https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/from/simultaneous-options/

2. Regarding "caps_lock -> command+control+option" rule, there are two mistakes:
    - `description` is inside the manipulator, but it should be above it (inside rule); just like the first rule.
    - The `to` event is wrong -- it only contains modifiers. *One* correct definition is:
    ```
    {
        "key_code": "left_command",
        "modifiers": [
            "left_control",
            "left_option"
        ]
    }
    ```

3. Using `shift` in `to_if_alone` may not work as you expected because it sends both press and release events at the same time. Actually, you don't need that, as the rule will only fire when you press both left and right `shift`s (because of `from.modifiers.mandatory`).
    - https://karabiner-elements.pqrs.org/docs/json/complex-modifications-manipulator-definition/to-if-alone/#about-keyboard-repeat

**Reply by Honestpuck** [reply]
Thank you. That got me working perfectly.
